lib/settings.ts

```diff
@@ -1,32 +1,59 @@
 import { DeepPartial } from 'utility-types';
+import { rules } from './rules/rule-index';
+
+const ruleNames = rules.map(r => r.ruleName);
 export interface ISettings {
   ports: {
     outRegex: string;
     inRegex: string;
-    enablePortStyle: boolean;
   };
   paths: {
     additional: string[];
     ignoreRegex: string;
   };
   style: {
-    preferredLogicTypePort: "unresolved" | "resolved" | "ignore";
-    preferredLogicTypeSignal: "unresolved" | "resolved" | "ignore";
+    preferredLogicTypePort: "unresolved" | "resolved";
+    preferredLogicTypeSignal: "unresolved" | "resolved";
```

Comment:
Why do you remove the possibility to ignore this? This gives more granular control over the style than disabling the whole rule.

Reply:
Why resolved?